### PR TITLE
update the example of custom memory allocator

### PR DESCRIPTION
**Pull request by wxyucs**

### CODE REVIEW

File: examples/cpp/custom_memory_allocator.cpp [hunk 1]
```diff
@@ -15,52 +15,75 @@
...
+        return addr;
     }
 
     void
     Deallocate(void* p) override {
+        if (sizes_.find(p) == sizes_.end())
+            return;
+        vsag::Options::Instance().logger()->Debug("deallocate " + std::to_string(sizes_[p]) +
+                                                  " bytes.");
+        sizes_.erase(p);
```
Comment (jiaweizone) [inline examples/cpp/custom_memory_allocator.cpp:42]: minor: reduce dup find cost
```C++
auto iter = sizes_.find(p);
if (iter == sizes_.end() {
    return;
}
sizes_.erase(iter);
```

**Review by jiaweizone — APPROVED** [review]
LGTM

File: examples/cpp/custom_memory_allocator.cpp (hunk 1 above)
Comment (wxyucs) [reply examples/cpp/custom_memory_allocator.cpp:42]: done

**Review by LHT129 — APPROVED** [review]
LGTM

**Pull request merged**